File: noop-java/src/main/java/com/exyui/android/debugbottle/components/injector/RunnableInjector.java
package com.exyui.android.debugbottle.components.injector;

import android.util.Log;

/**
 * Created by yuriel on 9/7/16.
 */

@SuppressWarnings("unused")
public class RunnableInjector {
    public final static RunnableInjector INSTANCE = new RunnableInjector();
    private static final String TAG = "FakeRunnableInjector";

    private RunnableInjector() {}

    public void put(Object a, Object b) {
        Log.d(TAG, "put");
    }

    public Runnable get() {
        return null;
    }

    public void run(Object a)  {
        Log.d(TAG, "run");
    }
}

File: noop/src/main/kotlin/com/exyui/android/debugbottle/components/injector/RunnableInjector.kt
package com.exyui.android.debugbottle.components.injector

import android.util.Log

/**
 * Created by yuriel on 8/30/16.
 */
object RunnableInjector {
    private val TAG = "FakeRunnableInjector"

    fun put(a: Any?, b: Any?) {
        Log.d(TAG, "add")
    }

    fun get(any: Any): Runnable? = null

    fun run(any: Any) {
        Log.d(TAG, "run")
    }

}